DragaliaAPI/Features/Dungeon/QuestEnemyService.cs

```diff
@@ -207,4 +216,45 @@ private IEnumerable<AtgenEnemy> GetEnemyList(int questId, int areaNum)
                 }
         );
     }
+
+    // Mercurial Gauntlet
+    private IEnumerable<AtgenEnemy> GetWallEnemyList(int wallId, int wallLevel)
+    {
+        QuestWallDetail questWallDetail = MasterAssetUtils.GetQuestWallDetail(wallId, wallLevel);
+        AreaInfo areaInfo = questWallDetail.AreaInfo.First();
+        logger.LogInformation("areaInfo: {@areaInfo}", areaInfo);
+
+        string assetName = $"{areaInfo.ScenePath}/{areaInfo.AreaName}".ToLowerInvariant();
+
+        if (!MasterAsset.QuestEnemies.TryGetValue(assetName, out QuestEnemies? enemies))
+        {
+            this.logger.LogWarning(
+                "Unable to retrieve enemy list for wall quest with wallId {wallId}, wallLevel {wallLevel}",
+                wallId,
+                wallLevel
+            );
+            return Enumerable.Empty<AtgenEnemy>();
+        }
+
+        if (!enemies.Enemies.TryGetValue(VariationTypes.Normal, out IEnumerable<int>? enemyList))
+        {
+            this.logger.LogWarning(
+                "Unable to retrieve enemy list for variation type {type}",
+                VariationTypes.Normal
+            );
+            return Enumerable.Empty<AtgenEnemy>();
+        }
```

Comment:
Is any of this actually needed if you just use `questWallDetail.BossEnemyParamId`

Reply:
ah when I wrote the code originally I just took it from the already existing enemy generating code, but it should just be as simple as generating a list of 1 enemy with the proper boss param id. Fixed in a recent commit